lib/OpenQA/WebSockets/Server.pm

```diff
@@ -112,70 +100,64 @@ sub check_authorized {
 }
 
 sub ws_create {
-    my ($self)   = @_;
+    my ($self) = @_;
     my $workerid = $self->param('workerid');
-    my $worker   = _validate_workerid($workerid);
-    unless ($worker) {
-        return $self->render(text => 'Unauthorized', status =>);
+    unless ($workers->{$workerid}) {
+        my $db = app->schema->resultset("Workers")->find($workerid);
+        unless ($db) {
+            return $self->render(text => 'Unauthorized', status =>);
```

Comment:
Touché.
There is this thing with replying some weird data on websockets we once discussed on IRC though. Can't remember and my server is still down, so can't access IRC logs, but @kraih had some comments too.
If it is possible, I'm for 401.